client/components/overflow-watcher/index.jsx

```diff
@@ -0,0 +1,61 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import { get } from 'lodash';
+
+const OverflowWatcher = Comp =>
+	class OverflowWatcherComponent extends Component {
+		state = {
+			overflowX: false,
+			ovewflowY: false,
+		};
+
+		handleResize = () => {
+			this.checkSize();
+		};
+
+		componentDidMount() {
+			window.addEventListener( 'resize', this.handleResize );
+			this.checkSize();
+		}
+
+		componentDidUpdate() {
+			this.checkSize();
+		}
+
+		componentWillUnmount() {
+			window.removeEventListener( 'resize', this.handleResize );
+		}
+
+		storeRef = node => {
+			this.sizeNode = get( node, [ 'refs', 'overflowNode' ], node );
+		};
+
+		checkSize = () => {
+			if ( ! this.sizeNode ) {
+				this.setState( {
+					overflowX: false,
+					overflowY: false,
+				} );
+			}
+
+			const domNode = findDOMNode( this.sizeNode );
+			const overflowX = domNode.scrollWidth > domNode.clientWidth + 4;
```

Comment:
ah, meant to move that to  a const. Just a buffer. I thought I saw some cases where clientHeight and scrollHeight will be different by a pixel. Might just pull it for now and see what happens.